From 37c670ad23d02b257f8646782d8a52a20397ab8b Mon Sep 17 00:00:00 2001
From: Zeng Zhaoming <b32542@freescale.com>
Date: Thu, 7 Apr 2011 15:59:04 +0800
Subject: [PATCH 0055/2508] ENGR00141859 Firmware: Fix Kernel build failed
 with firmware not find errors

Linux kernel build failed when source and output object separate store.

Fix it by add srctree to prefix firmware.

Signed-off-by: Zeng Zhaoming <b32542@freescale.com>
---
 firmware/Makefile |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/firmware/Makefile b/firmware/Makefile
index 8da933d..66d0e99 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -172,11 +172,12 @@ quiet_cmd_fwbin = MK_FW   $@
 		  ASM_WORD=$(if $(CONFIG_64BIT),.quad,.long);		     \
 		  ASM_ALIGN=$(if $(CONFIG_64BIT),3,2);			     \
 		  PROGBITS=$(if $(CONFIG_ARM),%,@)progbits;		     \
+		  FWPATH=$(if $(KBUILD_SRC),$(srctree)/)$(2);		     \
 		  echo "/* Generated by firmware/Makefile */"		> $@;\
 		  echo "    .section .rodata"				>>$@;\
 		  echo "    .p2align $${ASM_ALIGN}"			>>$@;\
 		  echo "_fw_$${FWSTR}_bin:"				>>$@;\
-		  echo "    .incbin \"$(2)\""				>>$@;\
+		  echo "    .incbin \"$${FWPATH}\""			>>$@;\
 		  echo "_fw_end:"					>>$@;\
 		  echo "   .section .rodata.str,\"aMS\",$${PROGBITS},1"	>>$@;\
 		  echo "    .p2align $${ASM_ALIGN}"			>>$@;\
-- 
1.7.7.4

